File: src/main/java/pranav/views/FloatingMenu/FMGroup.java
package pranav.views.FloatingMenu;

import android.content.Context;
import androidx.annotation.Nullable;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.preons.pranav.utilities.R;

import pranav.utilities.Utilities;

import static pranav.utilities.Animations.ANIMATION_DURATION;
import static pranav.utilities.Animations.DI;
import static pranav.utilities.Utilities.Resources.getColoredDrawable;

/**
 * Created on 09-06-2017 at 03:43 by Pranav Raut.
 * For QRCodeProtection
 */

@SuppressWarnings("unused")
final class FMGroup extends LinearLayout {
    private final boolean useCard;
    LayoutParams params = new LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    private FMGroupHelper groupDetail;
    private Context c = getContext();
    Utilities.Resources res = new Utilities.Resources(c);
    private LinearLayout container = new LinearLayout(c);
    private ImageView imageView = new ImageView(c);
    private TextView textView = new TextView(c);

    FMGroup(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        useCard = true;
        inti();
    }

    FMGroup(Context context, FMGroupHelper groupDetail) {
        super(context);
        this.groupDetail = groupDetail;
        useCard = groupDetail.isUseCard();
        inti();
    }

    private void inti() {
        setOrientation(VERTICAL);
        setLayoutParams(params);
        container();
        textView();
        imageView();
    }

    private void textView() {
        String s = groupDetail.getHeadTitle();
        if (s == null) textView.setVisibility(GONE);
        else {
            textView.setPadding(0, 0, 0, (int) res.getDimen(R.dimen.dPad));
            textView.setLayoutParams(params);
            textView.setText(s);
            textView.setTextColor(useCard ? 0x90000000 : 0x90ffffff);
            int i = (int) res.getDimen(R.dimen.dPad);
            if (useCard)
                textView.setPadding(i, textView.getPaddingTop(), i, textView.getPaddingTop());
        }
    }

    private void container() {
        container.setLayoutParams(params);
        container.setOrientation(VERTICAL);
    }

    private void imageView() {
        if (!groupDetail.isDividerVisible()) imageView.setVisibility(GONE);
        else {
            LayoutParams params1 = new LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, (int) res.getDimen(R.dimen.pad2dp));
            imageView.setImageDrawable(getColoredDrawable(res.getDimen(R.dimen.pad2dp), useCard ? 0x7f000000 : 0x7fffffff));
            imageView.setLayoutParams(params1);
        }
    }

    void addItem(FMItem item) {
        container.addView(item);
    }

    FMGroup add() {
        addView(textView);
        addView(container);
        addView(imageView);
        return this;
    }

    void initial() {
        textView.setScaleX(0);
        imageView.setScaleX(0);
    }

    void animateOpen() {
        textView.animate().scaleX(1).scaleY(1).setDuration(ANIMATION_DURATION)
                .setStartDelay(100)
                .setInterpolator(DI);
        imageView.animate().scaleX(1).setDuration(ANIMATION_DURATION)
                .setStartDelay(100)
                .setInterpolator(DI);
    }

    void animateClose() {
        textView.animate().scaleY(0).scaleX(0).setDuration(ANIMATION_DURATION).setInterpolator(DI);
        imageView.animate().scaleX(0).setDuration(ANIMATION_DURATION).setInterpolator(DI);
    }
}

File: src/main/java/pranav/views/floatingMenu/FMGroup.kt
package pranav.views.floatingMenu

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.preons.pranav.utilities.R
import pranav.utilities.Animations.ANIMATION_DURATION
import pranav.utilities.Animations.DI
import pranav.utilities.Utilities
import pranav.utilities.Utilities.Resources.getColoredDrawable

/**
 * Created on 09-06-2017 at 03:43 by Pranav Raut.
 * For QRCodeProtection
 */

class FMGroup : LinearLayout {
    private val useCard: Boolean
    private var params = LayoutParams(
        ViewGroup.LayoutParams.WRAP_CONTENT,
        ViewGroup.LayoutParams.WRAP_CONTENT
    )
    private lateinit var groupDetail: FMGroupHelper
    private val c = context
    var res = Utilities.Resources(c)
    private val container = LinearLayout(c)
    private val imageView = ImageView(c)
    private val textView = TextView(c)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        useCard = true
        inti()
    }

    constructor(context: Context, groupDetail: FMGroupHelper) : super(context) {
        this.groupDetail = groupDetail
        useCard = groupDetail.isUseCard
        inti()
    }

    private fun inti() {
        orientation = VERTICAL
        layoutParams = params
        container()
        textView()
        imageView()
    }

    private fun textView() {
        val s = groupDetail.headTitle
        if (s == null)
            textView.visibility = View.GONE
        else {
            textView.setPadding(0, 0, 0, res.getDimen(R.dimen.dPad).toInt())
            textView.layoutParams = params
            textView.text = s
            textView.setTextColor(if (useCard) -0x70000000 else -0x6f000001)
            val i = res.getDimen(R.dimen.dPad).toInt()
            if (useCard)
                textView.setPadding(i, textView.paddingTop, i, textView.paddingTop)
        }
    }

    private fun container() {
        container.layoutParams = params
        container.orientation = VERTICAL
    }

    private fun imageView() {
        if (!groupDetail.isDividerVisible)
            imageView.visibility = View.GONE
        else {
            val params1 = LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                res.getDimen(R.dimen.pad2dp).toInt()
            )
            imageView.setImageDrawable(
                getColoredDrawable(
                    res.getDimen(R.dimen.pad2dp),
                    if (useCard) 0x7f000000 else 0x7fffffff
                )
            )
            imageView.layoutParams = params1
        }
    }

    fun addItem(item: FMItem) {
        container.addView(item)
    }

    fun add(): FMGroup {
        addView(textView)
        addView(container)
        addView(imageView)
        return this
    }

    fun initial() {
        textView.scaleX = 0f
        imageView.scaleX = 0f
    }

    fun animateOpen() {
        textView.animate().scaleX(1f).scaleY(1f).setDuration(ANIMATION_DURATION)
            .setStartDelay(100).interpolator = DI
        imageView.animate().scaleX(1f).setDuration(ANIMATION_DURATION)
            .setStartDelay(100).interpolator = DI
    }

    fun animateClose() {
        textView.animate().scaleY(0f).scaleX(0f).setDuration(ANIMATION_DURATION).interpolator = DI
        imageView.animate().scaleX(0f).setDuration(ANIMATION_DURATION).interpolator = DI
    }
}
